src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -44,5 +44,26 @@ void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo)
 #endif
 }
 
+ICDRefreshKeyInfo * DefaultCheckInDelegate::OnKeyRefreshNeeded(const ICDClientInfo & clientInfo)
+{
+    if (mpICDRefreshKeyInfo != nullptr)
+    {
+        return nullptr;
+    }
+
+    mpICDRefreshKeyInfo = new ICDRefreshKeyInfo(this, clientInfo);
```

Comment:
when there are two check-in message that calls OnKeyRefreshNeeded, then collision happens

Reply:
Discussed offline and decided to make the variable local.